src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,197 @@
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node currentNode = indexIterator(index);
+        T tempValue = (T) currentNode.item;
+        currentNode.item = value;
+        return tempValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (size == 1) {
+            Node currentNode = head;
+            unlinkNode(currentNode, "zero_only");
+            return (T) currentNode.item;
+        }
+        if (size > 1 && index == 0) {
+            Node currentNode = head;
+            unlinkNode(currentNode, "head");
+            return (T) currentNode.item;
+        } else if (size > 1 && index == size - 1) {
+            Node currentNode = tail;
+            unlinkNode(currentNode, "tail");
+            return (T) currentNode.item;
+        }
+        Node currentNode = indexIterator(index);
+        unlinkNode(currentNode, "middle");
+        return (T) currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node currentNode = head;
+        if (size == 1 && equalityCheck(currentNode, object)) {
+            unlinkNode(currentNode, "zero_only");
+            return true;
+        }
+        if (size > 1 && equalityCheck(currentNode, object)) {
+            unlinkNode(currentNode, "head");
+            return true;
+        }
+        while (currentNode != null) {
+            if (currentNode.next != null && equalityCheck(currentNode, object)) {
+                unlinkNode(currentNode, "middle");
+                return true;
+            } else if (currentNode.next == null && equalityCheck(currentNode, object)) {
+                unlinkNode(currentNode, "tail");
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (size == 0);
+    }
+
+    private Node indexIterator(int index) {
+        if (index < size / 2) {
+            Node currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode;
+        }
+        Node currentNode = tail;
+        for (int i = size - 1; i > index; i--) {
+            currentNode = currentNode.prev;
+        }
+        return currentNode;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
+    }
+
+    private void checkIndexIfAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
+    }
+
+    private void unlinkNode(Node node, String nodePlace) {
```

Comment:
```suggestion
    private void unlinkNode(Node node) {
```
try to remove `nodePlace` and use checks for it